test/Faucet.t.sol

```diff
@@ -0,0 +1,158 @@
+    uint256 claimAmount;
+
+    function setUp() public override {
+        BaseSetup.setUp();
+
+        // deploy the faucet contract
+        faucetContract = new Faucet(
+            address(earlyAccessNFTAddress),
+            0.1 ether,
+            60 // 1 minute cooldown
+        );
+        claimAmount = faucetContract.claimAmount();
+
+        faucetContract.fundFaucet{value: 100 ether}();
+
+        uint256 balance = faucetContract.balance();
+        assertEq(balance, 100 ether);
+    }
+
+    // test claim
+    function test_ClaimShouldPass() public {
+        // mint whitelist nft to user1
+        earlyAccessNFTContract.mintTo(user1);
+
+        uint256 balanceBefore = address(user1).balance;
+        assertEq(balanceBefore, 0);
+
+        bool isSuccess = faucetContract.claim(payable(user1));
+        assertEq(isSuccess, true);
+
+        uint256 balanceAfter = address(user1).balance;
+        assertEq(balanceAfter, balanceBefore + claimAmount);
+    }
+
+    // test should revert if claimed before cool down
+    function test_ExpectRevertClaimTooSoon() public {
+        // mint whitelist nft to user1
+        earlyAccessNFTContract.mintTo(user1);
+
+        faucetContract.claim(payable(user1));
+        vm.expectRevert("Faucet: Claim too soon.");
+        faucetContract.claim(payable(user1));
+    }
+
+    // test should revert if account is not whitelisted
+    function test_ExpectRevertClaimAccountNotWhitelisted() public {
+        // no mint directly claim
+        vm.expectRevert("WhitelistNFT: Account is not whitelisted");
+        faucetContract.claim(payable(user1));
+    }
+
+    // test should pass when claim amount is set by owner
+    function test_SetAmountShouldPass() public {
+        faucetContract.setAmount(2);
+        uint256 amount = faucetContract.claimAmount();
+        assertEq(amount, 2);
+    }
+
+    // test should revert if claim amount is set by non owner
+    function test_ExpectRevertSetAmount() public {
+        vm.startPrank(user1);
+        vm.expectRevert();
+        faucetContract.setAmount(2);
+        vm.stopPrank();
+    }
+
+    // test should pass when cool down duration is set by owner
+    function test_SetCoolDownDurationShouldPass() public {
+        faucetContract.setDuration(2);
+        uint256 coolDownDuration = faucetContract.cooldownDuration();
+        assertEq(coolDownDuration, 2);
+    }
+
+    // test should revert if cool down duration is set by non owner
+    function test_ExpectRevertSetCoolDownDuration() public {
+        vm.startPrank(user1);
+        vm.expectRevert();
+        faucetContract.setDuration(2);
+        vm.stopPrank();
+    }
+
+    // test should pass when faucet is funded
+    function test_FundFaucetShouldPass() public {
+        uint256 balance = faucetContract.fundFaucet{value: 100 ether}();
+        assertEq(balance, 200 ether);
+    }
+
+    // test should revert if user claims twice before cool down
+    function test_ClaimTwiceShouldPass() public {
```

Comment:
Shouldn't this be ClaimTwiceShouldFail?

Reply:
ahh yes, thanks for pointing

Reply:
fixed [here](https://github.com/Karrot-Works/on-chain-whitelist/pull/5/commits/fde4901e374747baf4166d4095b14ba36175be29)